src/v/rpc/simple_protocol.cc

```diff
@@ -103,6 +103,17 @@ simple_protocol::dispatch_method_once(header h, net::server::resources rs) {
       = ssx::spawn_with_gate_then(
           rs.conn_gate(),
           [this, method_id, rs, ctx]() mutable {
+              if (ctx->get_header().version > transport_version::v0) {
+                  netbuf reply_buf;
+                  reply_buf.set_status(rpc::status::version_not_supported);
+                  reply_buf.set_version(transport_version::max_supported);
+                  return ctx->res.conn->input()
+                    .skip(ctx->get_header().payload_size)
+                    .then([ctx, reply_buf = std::move(reply_buf)]() mutable {
+                        return send_reply(ctx, std::move(reply_buf))
+                          .then([ctx]() mutable { ctx->signal_body_parse(); });
```

Comment:
nit: consider wrapping this in some `respond_and_skip_payload()` method and reusing it here and at L127

Reply:
nice idea. will do